src/enumerable.cr

```diff
@@ -1804,8 +1804,17 @@ module Enumerable(T)
   # (1..5).to_a # => [1, 2, 3, 4, 5]
   # ```
   def to_a
-    ary = [] of T
-    each { |e| ary << e }
+    to_a(&.itself)
+  end
+
+  # Returns an `Array` with all the elements returned from the *block*.
```

Comment:
The documentation should be closer to that of `#map` while highlighting their differences (and vice-versa in `#map`'s documentation too).

Reply:
Exactly. I would say that `map` is sometimes overwritten by some types (for example Tuple, Slice) to return the same type, and that if an `Array` is always needed then to use `to_a`.